lib/Ast_to_Mir.ml

```diff
@@ -281,16 +283,15 @@ let trans_prog filename
       | None | Some [] -> Skip
       | Some lst -> SList (List.map ~f:trans_stmt lst) )
   in
-  let lbind s = match s.stmt with SList ls -> ls | Skip -> [] | _ -> [s] in
+  let lbind s =
+    match s.stmt with SList ls | Block ls -> ls | Skip -> [] | _ -> [s]
+  in
   let coalesce stmts =
     let flattened = List.(concat (map ~f:lbind stmts)) in
-    with_no_loc (match flattened with [] -> Skip | _ :: _ -> SList flattened)
+    with_no_loc (SList flattened)
   in
   let pull_tvdecls_multi blocks =
-    let merge_maps maps =
-      List.map ~f:Map.Poly.to_alist maps
-      |> List.concat |> Map.Poly.of_alist_exn
-    and tvtables, stmts = blocks |> List.map ~f:pull_tvdecls |> List.unzip in
+    let tvtables, stmts = blocks |> List.map ~f:pull_tvdecls |> List.unzip in
```

Comment:
I'm pretty confused about what is going on here in `trans_prog`.

Why are you removing block structure from the statements? How do we make sure code gets generated for local variable scope then?

Reply:
Generally, why is there both a List of statements and Block of statements in the MIR?

Reply:
Blocks are a mechanism for local variable scoping. Lists are just lists of statements without such a mechanism. I'm not removing all blocks, just extra ones at the top level. It may have a bug and be removing one level too many, or something.

Reply:
Where are there extra blocks at the top level? Have looked at the code again and still don't understand why this `coalesce` is necessary. Could you explain a bit more the thinking behind it?

Reply:
The model block should be treated as being a local scope. Otherwise, all program blocks are global scope. Note however that a program block does not have a Block AST node at the top (even though it's written with curly braces).

I like that you merged the transformed parameters and model blocks in the MIR by the way. I hope we can do the same in the Stan language soon. In my mind, target should be available in transformed parameters and the model block should just be a local scope within transformed parameters. (At which point we should probably change the name of the compound block.)